packages/react-popup/Spec.md

```diff
@@ -1,31 +1,385 @@
+| leftCenter           | before        | center     |
+| leftBottomEdge       | before        | bottom     |
+| rightTopEdge         | after         | before     |
+| rightCenter          | after         | center     |
+| rightBottomEdge      | after         | bottom     |
+
+v8 uses `left` and `right`. v0 uses `before` and `after`. v0 vocabulary tries to be consistent regardless of RTL state. It's also possible to supply an explicit RTL hint to v8 which is a flip by default. v0 will flip by default but requires the consumer to detect RTL scenarios and modify props in these situations.
+
+In general the separation of both the position and alignment in v0 results in an API that is easier to use if a consumer only needs to modify one of the two props. However both try to achieve the same result in the end.
+
+It's important to note that if an incorrect pair of `position` and `align` are provided in v0, then `position` takes priority and `align` is set to `center`
+
+### Offset
+
+```tsx
+<ContextualMenu
+  // single number value
+  gap={100}
+/>
+
+<Popup
+  offset={[-100, 100]}
+/>
+
+// offset can also be a function of raw Popper properties
+const offsetFunction = ({
+  popper: PopperJs.Rect;
+  reference: PopperJs.Rect;
+  placement: PopperJs.Placement;
+}) => ([popper.width, -popper.height])
+```
+
+v8 positioning can only apply a numerical value to the first part position attribute of `DirectionalHint`. v0 supports a function to defer calculation at runtime. v0 also supports offset of the `Popup` in both axes while supporting RTL flips for offset values.
+
+### Bounds and overflow
+
+```typescript
+<Popup
+...
+  flipBoundary={htmlElement}
+  overflowBoundary={htmlElement}
+  mountNode={htmlElement}
+/>
+```
+
+v0 `Popup` provides 3 different properties to handle bounds and overflow:
+
+- flipBoundary - the bounds to calculate when to flip positioning of the popup
+- overflowBoundary - the bounds to shift the popup without overflowing
+- mountNode - where the popup is actually rendered in the DOM, by default this is a portal to a div in body
+
+```typescript
+<ContextualMenu
+  // pixel values for bounding rectangle
+  // defaults to target window as default bounding rectangle
+  bounds={{height: 0, width: 0, top: 0, left:0 , right: 0, bottom: 0}}
+  // callback for bounds
+  bounds{(target, targetWindow) => ({/*Same object as above*/})}
+  target={htmlElement}
+
+  // renders to a portal node on body
+  layerProps={/*ILayerProps*/}
+
+  // every single one of the above can all be declared here too
+  calloutProps={{bounds, target}}
+/>
+```
+
+v8 `Callout` has no notion of separate boundaries for flip or overflow, and auto behaviour is used for flip and overflow 'pushing'.
+
+### Events
+
+v8 provides the following positioning event callbacks
+
+- onLayerMounted -> proposed to be removed in converged Portal spec [#17824](https://github.com/microsoft/fluentui/pull/17824)
+- onPositioned
+- onScroll
```

Comment:
Might be good to add a little description for these (`onLayerMounted` and `onScroll` might be more self-descriptive but I feel like `onPositioned` is more obscure).